### PR TITLE
Add FXIOS-8935 - Display clear button when editing url

### PR DESCRIPTION
**Pull request by PARAIPAN9**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8935)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/19736)

## :bulb: Description

 - This PR adds the `Clear Button` when the `URL bar` is in editing state.
## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: SampleBrowser/SampleBrowser/Assets.xcassets/crossCircleFillLarge.imageset/Contents.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "images" : [
+    {
+      "filename" : "crossCircleFillLarge.pdf",
```
Comment (thatswinnie) [inline SampleBrowser/SampleBrowser/Assets.xcassets/crossCircleFillLarge.imageset/Contents.json:4]: This image has to be added to Client as well as the expectation is that every image added to `StandardImageIdentifiers` exists in Client as well.

File: BrowserKit/Sources/ToolbarKit/LocationView.swift [hunk 2]
```diff
@@ -39,11 +40,30 @@ class LocationView: UIView, UITextFieldDelegate, ThemeApplicable {
...
+    private var clearButtonWidthConstraint: NSLayoutConstraint?
+    private var gradientViewWidthConstraint: NSLayoutConstraint?
+
+    private lazy var clearButton: UIButton = .build { button in
+        button.setImage(
+            UIImage(named: StandardImageIdentifiers.Large.crossCircleFill)?.withRenderingMode(.alwaysTemplate),
+            for: .normal
+        )
+        button.addTarget(self, action: #selector(self.clearURLText), for: .touchUpInside)
+        button.accessibilityLabel = "Clean"
```
Comment (thatswinnie) [inline BrowserKit/Sources/ToolbarKit/LocationView.swift:52]: As this is in ToolbarKit we shall not use hard coded strings or a11y ids. These should be passed in through the `configure` method.

File: BrowserKit/Sources/ToolbarKit/LocationView.swift [hunk 3]
```diff
@@ -57,6 +77,12 @@ class LocationView: UIView, UITextFieldDelegate, ThemeApplicable {
         notifyTextChanged = { [self] in
             guard urlTextField.isEditing else { return }
 
+            if urlTextField.text?.isEmpty == true {
+                hideClearButton()
+            } else if urlTextField.text?.count == 1 {
```
Comment (thatswinnie) [inline BrowserKit/Sources/ToolbarKit/LocationView.swift:82]: `>= 1` instead of `== 1`?

Comment (PARAIPAN9) [reply BrowserKit/Sources/ToolbarKit/LocationView.swift:82]: This approach prevents the code from being called multiple times. If we used `>=`, the code would execute each time a character is typed.

Comment (thatswinnie) [reply BrowserKit/Sources/ToolbarKit/LocationView.swift:82]: Then what about the user pasting something? In that case the clear button wouldn't be shown.

File: BrowserKit/Sources/ToolbarKit/AddressToolbarState.swift [hunk 4]
```diff
@@ -6,8 +6,8 @@ import Foundation
 
 /// Defines the state for the address toolbar.
 public struct AddressToolbarState {
-    /// URL displayed in the address toolbar
-    let url: String?
+    /// VIew state for the `Location View`  in the address toolbar
```
Comment (thatswinnie) [inline BrowserKit/Sources/ToolbarKit/AddressToolbarState.swift:9]: Nit: 
```suggestion
    /// View state for the `Location View`  in the address toolbar
```

**Pull request merged**